test/fw/ptl/utils/pbs_crayutils.py

```diff
@@ -141,3 +141,36 @@ def count_node_state(self, state='UP  B'):
             if stat['State'] == state:
                 count += 1
         return count
+
+    def get_numthreads(self, nid):
+        """
+        Returns the number of hyperthread for the given node
+        """
+        options = '-N %d -n -f "nid,c/cu"' % int(nid)
+        hostname = socket.gethostname()
+        platform = self.du.get_platform(hostname)
+        apstat_env = os.environ
+        apstat_cmd = "apstat"
```

Comment:
Now that we have two functions that use apstat command we should probably separate out building of apstat command in its own function. These functions can then apply their options to the command and run it.
What do you think?

Reply:
That sounds like a good idea.  I'll try to do so.